### ISSUE TITLE
Crash: deadline below read timestamp is nonsensical

### ISSUE THREAD

**Issue by jkosh44** [cockroachdb/cockroach#96032]
**Describe the problem**

When using CockroachDB I encountered an unexpected error that caused a crash.

**To Reproduce**

I have not been able to reproduce this issue.

**Expected behavior**
The database not to crash.

**Additional data / screenshots**
[cockroach.log](https://github.com/cockroachdb/cockroach/files/10513787/cockroach.log)

Error:

> message: "internal error: deadline below read timestamp is nonsensical; txn has would have no chance to commit. Deadline: 1674753799.099895237,1. Read timestamp: 1674753799.099895237,2 Previous Deadline: 1674754043.251270841,0.",
>
> detail: "stack trace:
> github.com/cockroachdb/cockroach/pkg/kv/txn.go:775: UpdateDeadline()
> github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/leased_descriptors.go:220: maybeUpdateDeadline()
> github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/collection.go:145: MaybeUpdateDeadline()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:274: execStmtInOpenState()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129: func1()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2382: execWithProfiling()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:128: execStmt()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1920: func1()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1924: execCmd()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1846: run()
> github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:828: ServeConn()
> github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:728: func1()
> GOROOT/src/runtime/asm_amd64.s:1594: goexit()
> ",
> 
> hint: "You have encountered an unexpected error.
> 
> Please check the public issue tracker to check whether this problem is
> already tracked. If you cannot find it there, please report the error
> with details by creating a new issue.
> 
> If you would rather not post publicly, please contact us directly
> using the support form.
> 
> We appreciate your feedback.
> ",
> 
> position: None,
> where_: None,
> schema: None,
> table: None,
> column: None,
> datatype: None,
> constraint: None,
> file: "txn.go",
> line: 775,
> routine: "UpdateDeadline"

**Environment:**
 - CockroachDB versionCCL v22.2.0 @ 2022/12/05 16:37:36 (go1.19.1)
 - Server OS: Linux
 - Client app: rust-postgres

**Additional context**
Sorry that I do not have any more context. I was not able to reproduce the issue or even get a trace of all the queries that were ran. However, the error message made it seem like you'd rather I open an issue than not.

Jira issue: CRDB-23899

**Comment by benesch** [comment]
Possibly a duplicate of #76727? cc @ajwerner and @yuzefovich to assess (since they commented on the other issue).

**Comment by ajwerner** [comment]
Yeah, I'm happy with another issue, but I do think it duplicates https://github.com/cockroachdb/cockroach/issues/76727? I'm going to track there. This is high on the list. This shouldn't bubble up to the client, clearly there's a bug.

**Comment by ajwerner** [comment]
For what it's worth, I anticipate that the machine was severely overloaded. Do you think that's plausible?

**Comment by benesch** [comment]
> For what it's worth, I anticipate that the machine was severely overloaded. Do you think that's plausible?

Yep, definitely. Materialize is pretty Cockroach hungry, and the test job that failed runs several Materializes in parallel on the same machine as Cockroach.
